### DISCUSSION TITLE
Match file content for vim syntax highlighting

### DISCUSSION THREAD

**Discussion by reinerh** [netblue30/firejail#4319] [Show and tell]
Currently the provided syntax files for vim only detect profiles in `/etc/firejail/` and `~/.config/firejail/` (based on file extension).
For profile development it might be useful to let vim also detect the filetype in `firejail.git/etc/profile-*/`.

To accomplish this, you can add the following line to your `~/.vimrc`:

```vim
autocmd BufRead,BufNewFile * if getline(1) =~ '^# Firejail profile ' | setfiletype firejail | endif
```
(it will check the first line for `# Firejail profile`)

I haven't added it to `ftdetect/firejail.vim`, as I think it's not necessary to scan this for every file opened in vim.
But it might be interesting for people who develop profiles (@glitsj16 @rusty-snake :smile:).

**Comment by rusty-snake** [comment]
Thanks. Until now I allways `:set syntax=firejail` if I needed highlighting.

An other way is to add

```vim
autocmd BufNewFile,BufRead */firejail/*.inc setfiletype firejail
autocmd BufNewFile,BufRead */firejail/*.local setfiletype firejail
autocmd BufNewFile,BufRead */firejail/*.profile setfiletype firejail
```
 to `~/.vimrc` (based on https://gitlab.com/kmk3/dotfiles/-/blob/26e0ad5f95f2263e5cef8d1a2e7eca0f4c87bae5/src/.vimrc#L120-124).

**Comment by kmk3** [comment]
> Thanks. Until now I allways `:set syntax=firejail` if I needed highlighting.
> 
> An other way is to add
> 
> ```viml
> autocmd BufNewFile,BufRead */firejail/*.inc setfiletype firejail
> autocmd BufNewFile,BufRead */firejail/*.local setfiletype firejail
> autocmd BufNewFile,BufRead */firejail/*.profile setfiletype firejail
> ```
> 
> to `~/.vimrc` (based on https://gitlab.com/kmk3/dotfiles/-/blob/26e0ad5f95f2263e5cef8d1a2e7eca0f4c87bae5/src/.vimrc#L120-124).

Original code for reference:

```viml
autocmd BufNewFile,BufRead */firejail/*.inc set syntax=sh
autocmd BufNewFile,BufRead */firejail/*.local set syntax=sh
autocmd BufNewFile,BufRead */firejail/*.profile set syntax=sh
```

Related PR:

* <https://github.com/netblue30/firejail/pull/5850>
